src/components/StartPrjModal/CreateUsingProjectTemplate/index.tsx

```diff
@@ -0,0 +1,229 @@
+import Modal from 'components/Modal';
+import { FC, MouseEvent, useEffect, useState } from 'react';
+import getRandomString from 'utils/getRandomString';
+import clsx from 'clsx';
+import { useNavigate } from 'react-router-dom';
+import styles from './index.module.scss';
+import gif from './deleteme.gif';
+import CreatePrjModal from '../CreatePrjModal';
+import {
+  fetchAllProjectTemplates,
+  useCreateProjectFromTemplate,
+  useCreateTasksFromProjectTemplate,
+} from './hooks';
+
+interface ICreateUsingProjectTemplateProps {
+  onClose: () => void;
+  onNext: () => void;
+  orgId: number;
+}
+
+const CreateUsingProjectTemplate: FC<ICreateUsingProjectTemplateProps> = ({
+  onClose,
+  onNext,
+  orgId,
+}) => {
+  const navigate = useNavigate();
+  const [showProjectCreate, setShowProjectCreate] = useState(false);
+  const [currentTemplate, setCurrentTemplate] = useState<any>(null);
+  const [projectData, setProjectData] = useState<any>(null);
+
+  useEffect(() => {
+    if (projectData) {
+      createProjectTasksFromTemplate();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [projectData]);
+
+  const useFetchProjectTemplates = fetchAllProjectTemplates(setCurrentTemplate);
+  const createProjectFromTemplate =
+    useCreateProjectFromTemplate(setProjectData);
+  const createTasksFromProjectTemplate = useCreateTasksFromProjectTemplate();
+
+  const filterSelectedTemplate = (selectedTemplate: any) =>
+    setCurrentTemplate(selectedTemplate);
+
+  if (showProjectCreate) {
+    return <CreatePrjModal onClose={onClose} onNext={onNext} orgId={orgId} />;
+  }
+
+  const createTasksFromProjectTemplateFunc = () => {
+    if (!projectData) return [];
+    const temp: { [x: string]: any } = {};
+
+    const flProjectCategoryData = projectData.flProjectCategory.map(
+      (flProjectCategory: {
+        flProjectCategoryId: number;
+        categoryName: string;
+      }) => {
+        return {
+          flProjectCategoryTemplateId: flProjectCategory.flProjectCategoryId,
+          categoryName: flProjectCategory.categoryName,
+        };
+      }
+    );
+
+    currentTemplate.flProjectCategoryTemplate.forEach(
+      (flProjectCategoryTemplate: {
+        categoryName: string;
+        tasksTemplate: any;
+      }) => {
+        temp[flProjectCategoryTemplate.categoryName] =
+          flProjectCategoryTemplate.tasksTemplate;
```

Comment:
Wouldn't this override the previous value of `temp[flProjectCategoryTemplate.categoryName]`?

If there are 2 tasks of Dev category - Task 1 and Task 2
Then wouldn't this keep Task 2 as the latest value?